### PR TITLE
fixes #9120 - better feedback around composites for inc updates

### PR DESCRIPTION
**Pull request by jlsherrill**
* available incremental update api now returns the list of components
* UI displays them
* Inc update api will not accept composite version ids
* UI now sends up components properly

### CODE REVIEW

**Comment by waldenraines** [comment]
screen shot 2015-02-06 at 5 35 33 pm

Don't we want to show the new version of the component content view?  What about something like this where the user can click "Component Content Views" to expand/hide the views?

screen shot 2015-02-06 at 6 14 18 pm

I'm not a huge fan of this either solution either but I don't have any other ideas at the moment.

I think at the very least we want to show the versions that the components will be incremented to.  @ehelms do you have any thoughts on this screen?

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/errata/views/apply-errata-confirm.html [hunk 1]
```diff
@@ -49,7 +49,14 @@ <h3 ng-show="errata" translate>Apply {{ errata.errata_id }}</h3>
       <tbody>
         <tr ng-repeat="update in updates">
           <td>
-            <a ng-href="/content_views/{{ update.content_view_version.content_view.id }}/versions">{{ update.content_view_version.content_view.name }}</a>
+            <div>
+              <a ng-href="/content_views/{{ update.content_view_version.content_view.id }}/versions/{update.content_view_version.id}">{{ update.content_view_version.content_view.name }}</a>
```
Comment (waldenraines) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/errata/views/apply-errata-confirm.html]: Missing a set of brackets here: `{update.content_view_version.id}`

**Comment by jlsherrill** [comment]
@waldenraines i can add the next version.  I initially did exactly what you wrote, but thought it looked kinda bad.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/errata/views/apply-errata-confirm.html (hunk 1 above)
Comment (jlsherrill) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/errata/views/apply-errata-confirm.html]: Good catch :fish:

**Comment by waldenraines** [comment]
> @waldenraines i can add the next version. I initially did exactly what you wrote, but thought it looked kinda bad.

Hah, hmm.  Well I think ideally we'd show it in the table and somehow designate it's a component of the above composite view but I'm not sure of a great way to do this.

**Comment by jlsherrill** [comment]
@waldenraines added new version for components.  Also added an indicator with an explanation.  Let me know what you think.

**Comment by kybaker** [comment]
@waldenraines I modified the layout to include a larger row that will accommodate any detail we like about the components. I added a components label as well as an additional 'type' column. This should be more than enough flexibility and clarity without too much development overhead. Also adjusted some spacing and sizing.

incremental-update

**Comment by waldenraines** [comment]
Here is what we currently have in this PR for comparison's sake:

screen shot 2015-02-10 at 4 41 48 pm

I personally think it would look better without the `<li>` bullets and with the "Components" header above the list.  I also think that we could consider making them collapsable like @kybaker shows.  I don't like removing the striped table styles though so I'm glad you kept those.  So my suggestions is a blend of the two approaches.  Basically what @kybaker but maintaining the striped coloring of the table and removing the "Type" column.

I also think the sentence corresponding to the asterisk is oddly worded.  I think it should read something more like:

> These Content Views are Composite Content Views consisting of component Content Views listed underneath.

**Comment by waldenraines** [comment]
APJ

**Pull request merged**